new-site/src/components/Playground.js

```diff
@@ -0,0 +1,226 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useMDXScope } from 'gatsby-plugin-mdx/context';
+import { LiveProvider, LiveEditor, LiveError, LivePreview, withLive } from 'react-live';
+import { Tabs, TabList, TabPanel, Tab, Button, IconArrowUndo } from 'hds-react';
+import theme from 'prism-react-renderer/themes/github';
+
+import './Playground.scss';
+
+const Playground = ({ children }) => {
+  if (children) {
+    return <pre>{children}</pre>
+  }
+
+  return null;
+};
+
+Playground.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+export default Playground;
+
+const clearSelection = () => {
+  if (window.getSelection) {
+    window.getSelection().removeAllRanges();
+  } else if (document.selection) {
+    document.selection.empty();
+  }
+};
+
+const sanitize = (code) => {
+  const trimmedCode = code.trim();
+  return trimmedCode.startsWith('{') && trimmedCode.endsWith('}')
+    ? trimmedCode.substr(1, trimmedCode.length - 2).trim()
+    : trimmedCode;
+};
+
+const Editor = ({ onChange, initialCode, code, language }) => {
+  const viewPortRef = useRef();
+  const copyButtonRef = useRef();
+  const [resetCount, setResetCount] = useState(0);
+  const textAreaId = `code-block-textarea-${language}-${resetCount}`;
+  const helperTextId = `code-block-helper-${language}-${resetCount}`;
+  const getTextArea = useCallback((el) => el.querySelector(`#${textAreaId}`), [textAreaId]);
+
+  const onFocusKeyDown = useCallback(
+    (event) => {
+      if (event.key === 'Enter' && viewPortRef.current && event.target === viewPortRef.current) {
+        event.preventDefault();
+        const textArea = getTextArea(viewPortRef.current);
+        textArea.focus();
+      }
+    },
+    [getTextArea],
+  );
+
+  const onTextAreaBlur = useCallback((e) => {
+    // Set viewport focus on textarea Esc keypress (no related target)
+    if (viewPortRef.current && !e.relatedTarget) {
+      viewPortRef.current.focus();
+    }
+  }, []);
+
+  const copy = () => {
+    if (viewPortRef.current && copyButtonRef.current) {
+      const textArea = getTextArea(viewPortRef.current);
+      textArea.focus();
+      textArea.select();
+
+      try {
+        document.execCommand('copy');
+        if (copyButtonRef.current) {
+          copyButtonRef.current.focus();
+        }
+      } catch (err) {
+        console.warn('Oops, unable to copy');
+      }
+      clearSelection();
+    }
+  };
+
+  const reset = () => {
+    onChange(initialCode);
+    setResetCount(resetCount + 1);
+    if (viewPortRef.current) {
+      viewPortRef.current.focus();
+    }
+  };
+
+  useEffect(() => {
+    if (viewPortRef.current) {
+      const textArea = getTextArea(viewPortRef.current);
+      textArea.setAttribute('aria-describedby', helperTextId);
+      textArea.setAttribute('tabIndex', '-1');
+      textArea.addEventListener('blur', onTextAreaBlur);
+    }
+  }, [getTextArea, helperTextId, onTextAreaBlur]);
```

Comment:
I guess technically the `onTextAreaBlur` can never change in the current setting, as it's wrapped inside a `useCallback` hook that has an empty list as a dependency? But React probably still requires it to be set as a dependency for the useEffect.

Reply:
- [ ] simplify onTextAreaBlur